src/Education/CrossReferenceRuleHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Education
+{
+    /// <summary>
+    /// This is a helper class to encode a SonarCompositeRuleId into a valid URI and decode it vice versa.
+    /// </summary>
+    internal static class CrossReferenceRuleHelper
+    {
+        private const string prefix = "sonarlintrulecrossref";
+
+        public static Uri EncodeToUri(SonarCompositeRuleId compositeRuleId)
```

Comment:
We could make it an extension method, which would allow us to write code like
```
var uri = ruleId.EncodeToUri();

...

if (uri.TryGetRuleId(out ruleId)) ...
```

However, it's just syntactic sugar.
`

It would look a

Reply:
I think you forgot to finish the sentence - but regardless of that as I do not like extension methods because I think they are confusing so I will leave it as is